registration/include/pcl/registration/impl/transformation_estimation_eig.hpp

```diff
@@ -0,0 +1,238 @@
+  const pcl::PointCloud<PointSource> &cloud_src,
+  const std::vector<int> &indices_src,
+  const pcl::PointCloud<PointTarget> &cloud_tgt,
+  const std::vector<int> &indices_tgt,
+  Matrix4 &transformation_matrix) const
+{
+  if ( indices_src.size () != indices_tgt.size () )
+  {
+    PCL_ERROR ( "[pcl::TransformationEstimationEIG::estimateRigidTransformation] Number or points in source (%lu) differs than target (%lu)!\n", indices_src.size (), indices_tgt.size () );
+    return;
+  }
+
+  ConstCloudIterator<PointSource> source_it ( cloud_src, indices_src );
+  ConstCloudIterator<PointTarget> target_it ( cloud_tgt, indices_tgt );
+  estimateRigidTransformation ( source_it, target_it, transformation_matrix);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> void
+pcl::registration::TransformationEstimationEIG<PointSource, PointTarget, Scalar>::estimateRigidTransformation (
+  const pcl::PointCloud<PointSource> &cloud_src,
+  const pcl::PointCloud<PointTarget> &cloud_tgt,
+  const pcl::Correspondences &correspondences,
+  Matrix4 &transformation_matrix) const
+{
+  ConstCloudIterator<PointSource> source_it ( cloud_src, correspondences, true );
+  ConstCloudIterator<PointTarget> target_it ( cloud_tgt, correspondences, false );
+  estimateRigidTransformation ( source_it, target_it, transformation_matrix);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationEIG<PointSource, PointTarget, Scalar>::estimateRigidTransformation (
+  ConstCloudIterator<PointSource> &source_it,
+  ConstCloudIterator<PointTarget> &target_it,
+  Matrix4 &transformation_matrix) const
+{
+  // Convert to Eigen format
+  const int npts = static_cast <int> ( source_it.size () );
+
+  Eigen::Matrix3d sigma;
+  Eigen::Vector3d mean_s, mean_t;
+
+  mean_s.setZero ();
+  mean_t.setZero ();
+
+  for ( int i = 0; i < npts; ++i )
+  {
+    mean_s = mean_s + Eigen::Vector3d ( (double)source_it->x, (double)source_it->y, (double)source_it->z );
+    mean_t = mean_t + Eigen::Vector3d ( (double)target_it->x, (double)target_it->y, (double)target_it->z );
+    source_it++;
+    target_it++;
+  }
+  source_it.reset ();
+  target_it.reset ();
+
+  mean_s = mean_s / npts;
+  mean_t = mean_t / npts;
+
+  sigma.setZero ();
+
+  for ( int i = 0; i < npts; ++i )
+  {
+    Eigen::Vector3d source_v ( (double)source_it->x, (double)source_it->y, (double)source_it->z );
+    Eigen::Vector3d target_v ( (double)target_it->x, (double)target_it->y, (double)target_it->z );
+    sigma = sigma + ( target_v - mean_t ) * ( ( source_v - mean_s ).transpose () );
+    source_it++;
+    target_it++;
+  }
+  source_it.reset ();
+  target_it.reset ();
+  sigma = sigma / npts;

```

Comment:
This looks like a normal computation of mean and sigma. I wonder if PCL's existing codebase can be leveraged to reduce lines of code